webapp/angular.json

```diff
@@ -93,7 +93,13 @@
             "polyfills": "src/ts/polyfills.ts",
             "tsConfig": "tsconfig.spec.json",
             "karmaConfig": "tests/karma/karma-unit.conf.js",
-            "progress": true,
+            "codeCoverage": true,
+            "codeCoverageExclude": [
+              "src/js/bootstrapper/**",
+              "src/js/enketo/**",
+              "src/ts/environment/**",
+              "src/ts/polyfills.ts"
+            ],
```

Comment:
Why this exclusion? The code coverage report seems to cover only files under the `src/ts/` folder.

Reply:
The code coverage report only covers the /ts folder because of these exclusions.
It normally covers **all code** that exists in the bundle.

Reply:
Yes, but there are other files that don't fall in these rules, like `src/js/bootstrap-tour-standalone.js`, so I added it to the list in my local environment but didn't change the results of the report, is there another config I'm missing that causes that file not been covered?

Reply:
There is no other config. so I'm not sure why the tour is not covered.

Reply:
You can check the html report to see which files are indeed covered.

Reply:
Yes the rules under "codeCoverageExclude" definitely works because if I remove the section in my local environment, those files appear in the report.

I think the reason why the file does not appear and some others is because if the file is not covered by any test or called indirectly, eg. a file does not have a test but it is imported by another file that has tests, then the file is not included in the report at all. Eg. the file `src/ts/main.ts` does not appear in the report either.

Reply:
> Yes the rules under "codeCoverageExclude" definitely work because if I remove the section in my local environment, those files appear in the report.
> 
> I think the reason why the file does not appear and some others is because if the file is not covered by any test or called indirectly (eg. a file does not have a test but it is imported by another file that has tests), then the file is not included in the report at all. Eg. the file `src/ts/main.ts` does not appear in the report either.

Thanks for investigating the behavior here @mrsarm. I was puzzled by the same. After reading your comment I went and made a throwaway class under `/cht-core/webapp/src/ts/modals/`. Running the tests with that class never being used (by other code or tests) I do not see it on the test report. After that I made a random use of my throwaway class in a modal that is on the test report. After running the tests in that case I do see the throwaway class on the report.

I can't think of a case where this would be a problem, so there is probably no change to make here, but it's worth knowing the behavior.

Reply:
@mrsarm, your comment is right:
> the file is not covered by any test

The reporter config @ karma-unit.conf:

`skipFilesWithNoCoverage: true`
"Omits files with no statements, no functions and no branches covered from the report"

This simplifies the exclusions rules, thinking for example on the "route.ts" files, we don't need to worry about adding these exclusions at the moment.

Reply:
Let's see if by the end of the migration we can reach 75 on branches 😎 and then take more challenge

Reply:
Oops, I forgot about adding that setting. Do you guys think we should keep it or not?

Reply:
I like it :) simplifies the job. If devs get confused, we can always add an explanation in the project readme file for Devs.